### PR TITLE
Logs agent pipeline latency statistics on status page

### PR DESCRIPTION
**Pull request by gh123man**
### What does this PR do?

Expose logs agent pipeline latency in the status page. 
This change will show per source log latency in `ms` over the lifetime of the agent and within a 24h window. The specific implementation of tracking this data aims to be memory and CPU efficient as there could be a high volume of data. See the comments in `stats_tracker.go` for more details. `StatsTracker` is fairly generic and can be used for other types of aggregate tracking in other scenarios. 

### Motivation

AC-14

### Describe your test plan

Unit tests added. Manually tested with several logs source types.

### CODE REVIEW

File: pkg/logs/message/message.go [hunk 1]
```diff
@@ -5,26 +5,32 @@
...
+	Content            []byte
+	Origin             *Origin
+	status             string
+	IngestionTimestamp int64
 }
 
 // NewMessageWithSource constructs message with content, status and log source.
-func NewMessageWithSource(content []byte, status string, source *config.LogSource) *Message {
-	return NewMessage(content, NewOrigin(source), status)
+func NewMessageWithSource(content []byte, status string, source *config.LogSource, IngestionTimestamp int64) *Message {
```
Comment (remeh) [inline pkg/logs/message/message.go:23]: nit: lower camel-case for the parameter

File: pkg/logs/message/message.go [hunk 2]
```diff
@@ -5,26 +5,32 @@
...
 
 // Message represents a log line sent to datadog, with its metadata
 type Message struct {
-	Content []byte
-	Origin  *Origin
-	status  string
+	Content            []byte
+	Origin             *Origin
+	status             string
+	IngestionTimestamp int64
```
[the remark this answers is not in the document]
Comment (gh123man) [reply pkg/logs/message/message.go:19]: Actually it is used in the `tailer` when calling NewMessageWithSource.

File: pkg/util/stats_tracker_test.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	}
+	assert.Equal(t, int64(10), s.MovingAvg())
+
+	then = *now + 12*int64(time.Hour)
+
+	for *now < then {
+		s.Add(30)
+		*now += int64(time.Second)
+	}
+	// actually 19.99 but check for 19 because of truncation
```
Comment (ogaca-dd) [inline pkg/util/stats_tracker_test.go:78]: Note: What do you think about using https://godoc.org/github.com/stretchr/testify/assert#InDelta?

File: pkg/util/stats_tracker_test.go [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+	for *now < then {
+		s.Add(60)
+		*now += int64(time.Second)
+	}
+	// actually 44.99 but check for 44 because of truncation
+	assert.Equal(t, int64(44), s.MovingAvg())
+}
+
+// TestMovingAvg TODO
```
Comment (ogaca-dd) [inline pkg/util/stats_tracker_test.go:91]: Note : TODO

**Review by ogaca-dd — COMMENTED** [review]
LGTM, just a few comments (Feel free to ignore all `Note:`)

File: pkg/util/stats_tracker_test.go (hunk 3 above)
Comment (gh123man) [reply pkg/util/stats_tracker_test.go:78]: while that would be ideal, `s.MovingAvg()` returns an int64 so the truncation occurs internally. I'll clarify the comment instead.

File: pkg/status/templates/logsagent.tmpl [hunk 5]
```diff
@@ -52,24 +52,28 @@ Logs Agent
...
-      Inputs: {{ range $input := .inputs }}{{$input}} {{ end }}
-      {{- end }}
-      BytesRead: {{ .bytes_read }}
-      {{- if .info }}
-      Container Info:
-      {{- range $inf := .info }}
-        {{ $inf }}
-      {{- end }}
-      {{- end }}
+    Type: {{ .type }}
```
[the remark this answers is not in the document]
Comment (gh123man) [reply pkg/status/templates/logsagent.tmpl]: Nope - good catch!

**Review by ogaca-dd — APPROVED** [review]
LGTM

**Pull request merged**